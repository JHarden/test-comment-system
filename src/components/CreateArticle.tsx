import React from "react";

export interface CreateArticleProps {
    author: string;
}

class CreateArticle extends React.Component<CreateArticleProps, {}> {

    private title: string = '';
    private body: string = '';
    private votes: number = 1;

    private onSubmit = () => {
        console.log('submitted');
    }

    render() {
        const { author } = this.props;
        return(
            <div>  
                <div>by: {author} </div>
                <input type='text' name='title'></input>
                <div>
                    <input type='textArea' name='body'></input>
                </div>
                <button onClick={this.onSubmit}>Submit</button>
            </div>
        )
    }
}

export default CreateArticle;