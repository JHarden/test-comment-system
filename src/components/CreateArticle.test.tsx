import React from 'react';
import CreateArticle from './CreateArticle';
import renderer from 'react-test-renderer'
const mock = () => { }

const props = {
  author: 'john',
  text: 'this is a comment!',
  date: 'yesterday',
  onCreateArticle: () => {}
}

test('Create Article component test', () => {
  const component = renderer.create(
    <CreateArticle {...props}></CreateArticle>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
