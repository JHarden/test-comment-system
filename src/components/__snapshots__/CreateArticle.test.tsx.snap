// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Article component test 1`] = `
<div
  style={
    Object {
      "color": "blue",
      "outline": "1px solid blue",
    }
  }
>
  <input
    name="title"
    onChange={[Function]}
    type="text"
  />
  <div>
    <textarea
      name="body"
      onChange={[Function]}
    />
  </div>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;

exports[`RaceSelector component test 1`] = `
<div
  style={
    Object {
      "color": "blue",
      "outline": "1px solid blue",
    }
  }
>
  <input
    name="title"
    onChange={[Function]}
    type="text"
  />
  <div>
    <textarea
      name="body"
      onChange={[Function]}
    />
  </div>
  <button
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
